### ISSUE TITLE
Cannot launch two dependencies of the same "given name" even after destroying reference of first one

### ISSUE THREAD

**Issue by nathan-murphy** [JKISoftware/JKI-State-Machine-Objects#72]
**Describe the bug**
Stopping a dependency does not clear its Name from the FQN Array, so if you try and launch any subsequent dependency it will have a different name from the one given. This would be fine, but the "FindDependency" function using Name as key will only look at the qualified names, not the given names. 

**To Reproduce**
1. Create an SMO (owner)
2. Launch a Dependent SMO named Child
3. Stop the Dependency and destroy the reference
4. Re-launch a Dependent SMO named Child
5. Try and locate that Dependency using the "FindDependency" using Name as the Key
6. Get error 42 because the dependency can't be found

**Expected behavior**
I would expect FindDependency using Name as key to search the "Given Name" instead of the FQN.

**Additional context**
Using the 1.4.0_beta branch

**Comment by francois-normandin** [comment]
Hi @nathan-murphy, 

I've tried to reproduce the issue you describe, and the closest I get is to generate a warning "42" when I try to stop and destroy the reference while not "owning" the reference. However, I can get the child to be launched again, and found by name. 

My test was performed in SMO 1.3.0 build 56 (latest release), so I've attached the test code for you to run and compare on 1.4beta.

Are you perhaps using the Public API's Stop/Destroy commands? If so, this is a "no-op" call because only the owner can stop and destroy a dependency's reference.  If this is the case, you would be creating a second object named Child, which takes a different name. ("Child__001")

**Here's a screenshot of the test VI I created. Please let me know if this captures the essence of your issue.**
Since the dynamic dependency gets "owned" by the caller, only the caller can perform a stop/destroy state change request. To ensure that the owner manages the lifetime of its dependency, only a protected method can be called to terminate the dependency. (Under the hood, it provides the "shared key" that verifies ownership relation)

![image](https://user-images.githubusercontent.com/11728548/107590862-48e46380-6bd7-11eb-876c-5a1a5e97389e.png)

If it does not work in 1.4 beta, then something changed and it's a bug introduced after 1.3.0
If it is working, then I assume my transposition of your report in test code is missing the target. Maybe you can provide example code that reproduces the issue?

best,
Francois
[SMO Issue 72.zip](https://github.com/JKISoftware/JKI-State-Machine-Objects/files/5962572/SMO.Issue.72.zip)


_PS. Please be aware that the beta version 1.4.x might still not have all its features make it to production release. There are configuration management features that are still not fully vetted, or desired as presented. Any feedback on those features would be appreciated._

**Comment by nathan-murphy** [comment]
Thanks for the additional context.

I tried running this in the 1.4 beta - here's my front panel output:
![image](https://user-images.githubusercontent.com/10726539/107645908-f474ce80-6c3e-11eb-946f-d38359761ab0.png)

Some notes - I modified the SMO base class that is present in the 1.4 branch for the following modifications:

- Added shift registers to the SMO and error terminals in the base class Process VI
- Added some default code in the onDependencyMessage (to execute the 'do' method of any Dependency Message.lvclass that is passed in from a dependency
- Added another protected VI to wrap the NotifyOwner
- Moved all SMO content into an lvlib, built into an lvlibp to support plugins ([this ](https://forums.ni.com/t5/LabVIEW-Development-Best/Building-a-basic-plugin-based-application-with-HAL-using-JKI-SMO/tac-p/4127589?profile.language=en)is what I was working on when I found this...)


This was all done in LabVIEW 2020. I'll include my source code as well.
[SMO Issue 72 using Nate's 1.4.zip](https://github.com/JKISoftware/JKI-State-Machine-Objects/files/5965976/SMO.Issue.72.using.Nate.s.1.4.zip)
